src/script/script_scanner.cpp

```diff
@@ -198,8 +198,8 @@ struct ScriptFileChecksumCreator : FileScanner {
 static bool IsSameScript(const ContentInfo &ci, bool md5sum, ScriptInfo *info, Subdirectory dir)
 {
 	uint32_t id = 0;
-	const char *str = info->GetShortName().c_str();
-	for (int j = 0; j < 4 && *str != '\0'; j++, str++) id |= *str << (8 * j);
+	std::string str = info->GetShortName().substr(0, 4);
+	for (size_t j = 0; j < str.size(); j++) id |= str[j] << (8 * j);
```

Comment:
Well, this was broken before. I guess scripts rarely use chars >= 0x80.
```suggestion
	for (size_t j = 0; j < str.size(); j++) id |= static_cast<uint8_t>(str[j]) << (8 * j);
```